### PR TITLE
ENT-1887: Port UserResource to spec-first

### PR DESCRIPTION
**Pull request by Januson**

### CODE REVIEW

File: api/candlepin-api-spec.yaml [hunk 1]
```diff
@@ -2414,6 +2414,292 @@ paths:
...
+                  $ref: '#/components/schemas/UserDTO'
+        default:
+          $ref: '#/components/responses/default'
+    post:
+      description: Creates a User
+      tags:
+        - Users
+      operationId: createUser
+      requestBody:
+        description: The updated user
```
[the remark this answers is not in the document]
Comment (Januson) [reply api/candlepin-api-spec.yaml]: Done.

File: api/candlepin-api-spec.yaml [hunk 2]
```diff
@@ -2414,6 +2414,292 @@ paths:
...
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UserDTO'
+      security: []
+      responses:
+        200:
+          description: The created user
```
[the remark this answers is not in the document]
Comment (Januson) [reply api/candlepin-api-spec.yaml]: Done.

File: server/src/main/java/org/candlepin/resource/UserResource.java [hunk 3]
```diff
@@ -69,22 +54,20 @@
...
 @Api(value = "users", authorizations = { @Authorization("basic") })
-public class UserResource {
-
-    private UserServiceAdapter userService;
-    private I18n i18n;
-    private OwnerCurator ownerCurator;
-    private ModelTranslator modelTranslator;
+public class UserResource implements UsersApi {
 
+    private final UserServiceAdapter userService;
```
[the remark this answers is not in the document]
Comment (Januson) [reply server/src/main/java/org/candlepin/resource/UserResource.java:59]: It's not that it's needed. Just something I like to do. By making them final together with adding invariant checks such as null validation to the constructor we can make sure that after the constructor completes we have a fully initialized object.

[the remark this answers is not in the document]
Comment (sonalidhome) [reply server/src/main/java/org/candlepin/resource/UserResource.java:59]: Gotcha.

**Pull request merged**